src/components/nav/nav.tsx

```diff
@@ -0,0 +1,201 @@
+  }
+})
+
+/**
+ * @slot logo - A slot for adding a `calcite-logo` component to the primary nav level.
+ * @slot user - A slot for adding a `calcite-user` component to the primary nav level.
+ * @slot progress - A slot for adding a `calcite-progress` component to the primary nav level.
+ * @slot menu-action - A slot for adding a `calcite-action` component to the primary nav level.
+ * @slot primary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the primary nav level.
+ * @slot primary-content-center - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the center position of the primary nav level.
+ * @slot primary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the primary nav level.
+ * @slot secondary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the secondary nav level.
+ * @slot secondary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the secondary nav level.
+ * @slot tertiary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the tertiary nav level.
+ * @slot tertiary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the tertiary nav level.
+ */
+export class CalciteNav {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteNavElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, display a hamburger `calcite-action` and emits `calciteNavMenuActionSelect` event on user interaction.
+   */
+  @Prop({ reflect: true }) navAction = false;
+
+  /**
+   * Specifies the label of the hamburger icon.
+   */
+  @Prop() label!: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  @State() logoSlotHasElements: boolean;
+
+  @State() userSlotHasElements: boolean;
+
+  @State() primarySlotHasElements: boolean;
+
+  @State() secondarySlotHasElements: boolean;
+
+  @State() teritiarySlotHasElements: boolean;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Emits whenever the component is selected or unselected.
```

Comment:
> When `navAction` is `true`, emits when the component's selection changes.

Reply:
Good point, how about a reference to the `calcite-action` for the second reference? Perhaps something like:

> When `navAction` is `true`, emits when the `calcite-action` selection changes.

Reply:
i think its better to mention `calcite-action` **menu** since user can slot calcite-action in any of the slots available and assume selection of those will emit the event.

Reply:
I still don't think we want to use `menu` with this property - it could be used to launch something without a menu. This would be the only action we are rendering from inside the `nav` component.

What about:

`When `navAction` is true, emits when the user interacts with the displayed action.`

Reply:
i think menu makes sense since we are displaying menu kind of icon. This can help eliminate the confusion between other slotted elements which can be `calcite-action`'s . And the event is emitted only on user interaction with hamburger icon.

Reply:
updated to 
```
When `navAction` is `true`, emits when the displayed action selection changes.
```